components/dashboard/ActivityInputDialog.js

```diff
@@ -0,0 +1,356 @@
+    setSubmitting(false);
+    setFormValues(activityFormValues);
+    setFormErrors({});
+  };
+
+  return (
+    <Dialog
+      fullWidth
+      onClose={onClose}
+      aria-labelledby="customized-dialog-title"
+      open={show}
+      onExited={resetComponent}
+    >
+      <DialogTitle id="customized-dialog-title" onClose={onClose}>
+        <Typography variant="h5">Add Activity</Typography>
+      </DialogTitle>
+      <DialogContent dividers>
+        {submitting && <CircularProgress />}
+        {!(submitting || success) && (
+          <form className={classes.container} id={formId}>
+            <FormControl className={classes.formControl}>
+              <InputLabel id={`${formId}-activityType`}>Activity</InputLabel>
+              <Select
+                labelId={`${formId}-activityType`}
+                id="activityType-select"
+                value={formValues.activityType}
+                onChange={e => setFormValues({ ...formValues, activityType: e.target.value })}
+              >
+                {ACTIVITY_TYPES.map(activity => (
+                  <MenuItem key={activity} value={activity}>
+                    {activity}
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+            <FormControl className={classes.formControl}>
+              <InputLabel shrink htmlFor="description-textfield">
+                Description
+              </InputLabel>
+              <TextField
+                id="description-textfield"
+                value={formValues.description}
+                error={formErrors && formErrors.description}
+                helperText={formErrors && formErrors.description ? formErrors.description : ''}
+                fullWidth
+                placeholder=" "
+                onChange={e => setFormValues({ ...formValues, description: e.target.value })}
+                className={classes.textField}
+              />
+            </FormControl>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <Grid container className={classes.formControl}>
+                <KeyboardDatePicker
+                  disableToolbar
+                  disableFuture
+                  variant="inline"
+                  format="MM/dd/yyyy"
+                  margin="normal"
+                  id={`${formId}--dateCompleted`}
+                  label="Date Completed"
+                  value={formValues.dateCompleted}
+                  onChange={date => setFormValues({ ...formValues, dateCompleted: date })}
+                  KeyboardButtonProps={{
+                    'aria-label': 'change date',
+                  }}
+                />
+              </Grid>
+            </MuiPickersUtilsProvider>
+            <FormControl className={classes.formControl}>
+              <InputLabel id={`${formId}-timeSpent`}>Time Spent</InputLabel>
+              <Select
+                labelId={`${formId}-timeSpent`}
+                id="timeSpent-select"
```

Comment:
Could we include `formId` in this (and a few others), just to keep the IDs "namespaced"?

Also note the default `customized-dialog-title`